zebrad/tests/config.rs

```diff
@@ -0,0 +1,342 @@
+    original_vars: Vec<(String, String)>,
+}
+
+impl EnvGuard {
+    /// Acquire the global lock and clear all ZEBRA_* env vars, saving originals.
+    fn new() -> Self {
+        // If a test panics, the mutex guard is dropped, but the mutex remains poisoned.
+        // We can recover from the poison error and get the lock, because we're going
+        // to overwrite the environment variables anyway.
+        let guard = TEST_MUTEX.lock().unwrap_or_else(|e| e.into_inner());
+
+        let original_vars: Vec<(String, String)> = env::vars()
+            .filter(|(key, _val)| key.starts_with("ZEBRA_"))
+            .collect();
+
+        for (key, _) in &original_vars {
+            env::remove_var(key);
+        }
+
+        Self {
+            _guard: guard,
+            original_vars,
+        }
+    }
+
+    /// Set a ZEBRA_* environment variable for this test.
+    fn set_var(&self, key: &str, value: &str) {
+        env::set_var(key, value);
+    }
+}
+
+impl Drop for EnvGuard {
+    fn drop(&mut self) {
+        // Clear any ZEBRA_* set during the test
+        let current_vars: Vec<String> = env::vars()
+            .filter(|(key, _)| key.starts_with("ZEBRA_"))
+            .map(|(key, _)| key)
+            .collect();
+        for key in current_vars {
+            env::remove_var(&key);
+        }
+
+        // Restore originals
+        for (key, value) in &self.original_vars {
+            env::set_var(key, value);
+        }
+    }
+}
+
+// --- Defaults and file loading ---
+
+#[test]
+fn config_load_defaults() {
+    let _env = EnvGuard::new();
+
+    let config = ZebradConfig::load(None).expect("Should load default config");
+
+    assert_eq!(config.network.network.to_string(), "Mainnet");
+    assert_eq!(config.rpc.listen_addr, None); // RPC disabled by default
+    assert_eq!(config.metrics.endpoint_addr, None); // Metrics disabled by default
+}
+
+#[test]
+fn config_load_from_file() {
+    let _env = EnvGuard::new();
+
+    let temp_dir = TempDir::new().expect("create temp dir");
+    let config_path = temp_dir.path().join("test_config.toml");
+
+    let test_config = r#"
+[network]
+network = "Testnet"
+
+[rpc]
+listen_addr = "127.0.0.1:8232"
+
+[metrics]
+endpoint_addr = "127.0.0.1:9999"
+"#;
+
+    fs::write(&config_path, test_config).expect("write test config");
+
+    let config = ZebradConfig::load(Some(config_path)).expect("load config from file");
+
+    assert_eq!(config.network.network.to_string(), "Testnet");
+    assert_eq!(
+        config.rpc.listen_addr.unwrap().to_string(),
+        "127.0.0.1:8232"
+    );
+    assert_eq!(
+        config.metrics.endpoint_addr.unwrap().to_string(),
+        "127.0.0.1:9999"
+    );
+}
+
+#[test]
+fn config_nonexistent_file_errors() {
+    let _env = EnvGuard::new();
+
+    let nonexistent_path = PathBuf::from("/this/path/does/not/exist.toml");
+    let result = ZebradConfig::load(Some(nonexistent_path));
+    assert!(result.is_err(), "Should fail to load nonexistent file");
```

Comment:
Optional/Nitpick: We can use `.expect_err()` here and below where we don't want to match on a specific error value.

```suggestion
    let _env = EnvGuard::new();
    ZebradConfig::load(Some(PathBuf::from("/this/path/does/not/exist.toml")))
        .expect_err("should fail to load nonexistent file");
```

Reply:
Solved here, for multiple cases [`594874a` (#9768)](https://github.com/ZcashFoundation/zebra/pull/9768/commits/594874a50f00b4cf1f20e8dddeda95007b012ebe)